spring-cloud-azure-autoconfigure/src/main/java/com/microsoft/azure/spring/cloud/autoconfigure/telemetry/TelemetryTracker.java

```diff
@@ -31,55 +32,69 @@
 
     private static final String PROPERTY_SERVICE_NAME = "serviceName";
 
+    private static final String SPRING_CLOUD_AZURE_EVENT = "spring-cloud-azure";
+
     private static final String TELEMETRY_INVALID_KEY = "invalid-instrumentationKey";
 
     private static final int INSTRUMENTATION_KEY_LENGTH = 36;
 
-    private static final Logger LOG = LoggerFactory.getLogger(TelemetryTracker.class);
-
     private final TelemetryClient client;
 
     private final Map<String, String> defaultProperties;
 
-    public TelemetryTracker(Azure azure, String resourceGroup, TelemetryProperties telemetryProperties) {
-        this.client = this.getTelemetryClient(telemetryProperties);
+    public TelemetryTracker(@NonNull String subscriptionId, @NonNull String resourceGroup, String instrumentationKey) {
+        this.defaultProperties = buildDefaultProperties(subscriptionId, resourceGroup);
+        this.client = buildTelemetryClient(instrumentationKey);
+    }
+
+    public static void triggerEvent(TelemetryTracker tracker, String serviceName) {
+        if (tracker != null) {
+            tracker.trackEventWithServiceName(serviceName);
+        }
+    }
+
+    private Map<String, String> buildDefaultProperties(String subscriptionId, String resourceGroup) {
+        Map<String, String> properties = new HashMap<>();
 
-        this.defaultProperties =  new HashMap<>();
+        properties.put(PROPERTY_SUBSCRIPTION_ID, subscriptionId);
+        properties.put(PROPERTY_RESOURCE_GROUP, resourceGroup);
+        properties.put(PROPERTY_VERSION, PROJECT_INFO);
+        properties.put(PROPERTY_INSTALLATION_ID, MacAddressHelper.getHashedMacAddress());
 
-        this.defaultProperties.put(PROPERTY_SUBSCRIPTION_ID, azure.getCurrentSubscription().subscriptionId());
-        this.defaultProperties.put(PROPERTY_RESOURCE_GROUP, resourceGroup);
-        this.defaultProperties.put(PROPERTY_VERSION, PROJECT_INFO);
-        this.defaultProperties.put(PROPERTY_INSTALLATION_ID, TelemetryUtils.getHashMac());
+        return Collections.unmodifiableMap(properties);
     }
 
-    private TelemetryClient getTelemetryClient(TelemetryProperties telemetryProperties) {
-        final TelemetryClient client = new TelemetryClient();
-        final String instrumentationKey = telemetryProperties.getInstrumentationKey();
+    private TelemetryClient buildTelemetryClient(String instrumentationKey) {
+        TelemetryClient client = new TelemetryClient();
 
-        if (StringUtils.hasText(instrumentationKey) && instrumentationKey.length() == INSTRUMENTATION_KEY_LENGTH) {
-            client.getContext().setInstrumentationKey(instrumentationKey);
-        } else {
-            client.getContext().setInstrumentationKey(TELEMETRY_INVALID_KEY);
+        if (!isValid(instrumentationKey)) {
             LOG.warn("Telemetry instrumentationKey {} is invalid", instrumentationKey);
+            throw new RuntimeException("Telemetry instrumentationKey is invalid");
```

Comment:
From my side, I don't think it should throw runtime exception here.

1. telemetry is for our BI data, in fact, we three are almost the only users for this module.
2. Too easy for this runtime exception, if config BI allowed, and without this key in maven command line will trigger this.
3. For any other users, they do not honor this, all about telemetry to them is `telemetryAllowed`.

Reply:
1. Whether this class is used internally or externally, we should throw such exception since without correct key, this instance is useless. We need to throw it to let user to be aware of it.
2. Yes. So we might need catch this exception in other place. In this PR, I just merged the conflict. 
3. Yes. This key is transparent for end users. Even user build the project without this, this should run clean.

Reply:
Well, I am OK with exception, my point is that there must be trade-off somewhere. From my side, involve exception looks too heavy here.